run-infracost/action.yml

```
@@ -0,0 +1,68 @@
+name: run-infracost
+description: Runs infracost to get a cost estimate.
+
+inputs:
+  plan-file:
+    description: Path to saved terraform plan file
+    required: true
+  working-directory:
+    description: Working directory, in which the infracost command is run
+    required: true
+  infracost-api-endpoint:
+    description: Self-hosted api-endpoint
+    required: false
+    default: https://infracost.pafcloud.net
+
+outputs:
+  output:
+    description: infracost output (diff)
+    value: ${{ steps.this.outputs.output }}
+  summary:
+    description: Short summary containing old monthly + new monthly
+    value: ${{ steps.this.outputs.summary }}
+
+runs:
+  using: composite
+  steps:
+    - id: this
+      shell: bash
+      working-directory: ${{ inputs.working-directory }}
+      env:
+        INFRACOST_CURRENCY: EUR
+        INFRACOST_CURRENCY_FORMAT: "EUR: 1.234,56€"
+        INFRACOST_PRICING_API_ENDPOINT: ${{ inputs.infracost-api-endpoint }}
+        PLAN_FILE: ${{ inputs.plan-file }}
+      run: |
+        if [ -z "$GITHUB_WORKSPACE" ]; then
+          echo "ERROR: environment variable GITHUB_WORKSPACE is unset"
+          exit 1
+        fi
+
+        search_stop_dir="$(dirname $GITHUB_WORKSPACE)"
+        upfind() {
+          file="$1"
+          while [ "$PWD" != "$search_stop_dir" -a "$PWD" != "/" ]; do
+            if [ -r "$file" ]; then
+              echo "$file"
+              break
+            fi
+            cd ..
+          done
+        }
+
+        usage_file="$(upfind resource-use.yml)"
```

```suggestion
        usage_file="$(upfind infracost-usage.yml)"
```
per [example](https://www.infracost.io/docs/features/usage_based_resources/#3-run-with-usage-file)